app/src/main/java/com/google/jetpackcamera/MainActivity.kt

```diff
@@ -126,7 +128,15 @@ class MainActivity : ComponentActivity() {
 
     private fun getPreviewMode(): PreviewMode {
         if (intent == null || MediaStore.ACTION_IMAGE_CAPTURE != intent.action) {
-            return PreviewMode.StandardMode
+            return PreviewMode.StandardMode { event ->
+                if (event is PreviewViewModel.ImageCaptureEvent.ImageSaved) {
+                    val intent = Intent(Camera.ACTION_NEW_PICTURE)
+                    if (event.savedUri != null) {
```

Comment:
OutputFileResults.savedUri is nullable. This makes it still broadcast in that case, just with no data

Reply:
According to the documentation for `OutputFileResults.savedUri`:

> Returns null if the [OutputFileOptions](https://developer.android.com/reference/androidx/camera/core/ImageCapture.OutputFileOptions) is constructed with [#Builder(OutputStream)](https://developer.android.com/reference/androidx/camera/core/ImageCapture.OutputFileOptions.Builder).

AFAIK, we only use `Builder(OutputStream)` for the external image capture flow. However, in that case we know that the URI won't be null, since it's passed in. So in both cases, we should never actually have a null URI. We should write the code accordingly to expect that we never have a null URI. This may require moving away from returning a `OutputFileResults` from `takePicture`, and just return the URI directly.